crates/turborepo-lib/src/shim.rs

```diff
@@ -792,306 +724,149 @@ mod test {
-                        path: project_one.clone(),
-                        has_package_json: true,
-                        has_turbo_json: true,
-                        workspace_globs: None,
-                    },
-                ],
-                output: Ok(project_two.clone()),
-            },
-            // The below test ensures that we privilege a valid `turbo.json` structure prior to
-            // evaluation of a valid `package.json` structure. If you include `turbo.json` you are
-            // able to "skip" deeper into the resolution by disregarding anything that does _not_
-            // have a `turbo.json`. This will matter _far_ more in a multi-language environment.
-
-            // Just one example test proves that the entire alternative chain construction works.
-            // The selection logic from within this set is identical. If we attempt to optimize the
-            // number of file system reads by early-exiting for matching we should expand this test
-            // set to mirror the above section.
-            TestCase {
-                description: "Nested non-monorepo packages, turbo.json primacy.",
-                infer_infos: vec![
-                    InferInfo {
-                        path: project_two,
-                        has_package_json: true,
-                        has_turbo_json: false,
-                        workspace_globs: None,
-                    },
-                    InferInfo {
-                        path: project_one.clone(),
-                        has_package_json: true,
-                        has_turbo_json: true,
-                        workspace_globs: None,
-                    },
-                ],
-                output: Ok(project_one),
-            },
+            (&irrelevant, None),
+            (
+                &monorepo_root,
+                Some(RepoState {
+                    root: monorepo_root.clone(),
+                    mode: RepoMode::MultiPackage,
+                }),
+            ),
+            (
+                &app_1,
+                Some(RepoState {
+                    root: monorepo_root.clone(),
+                    mode: RepoMode::MultiPackage,
+                }),
+            ),
+            (
+                &app_1_src,
+                Some(RepoState {
+                    root: monorepo_root.clone(),
+                    mode: RepoMode::MultiPackage,
+                }),
+            ),
+            (
+                &single_root,
+                Some(RepoState {
+                    root: single_root.clone(),
+                    mode: RepoMode::SinglePackage,
+                }),
+            ),
+            (
+                &single_root_src,
+                Some(RepoState {
+                    root: single_root.clone(),
+                    mode: RepoMode::SinglePackage,
+                }),
+            ),
+            // Nested, technically not supported
+            (
+                &standalone,
```

Comment:
Don't understand this one. Why is `standalone` not just a workspace in `monorepo_root`? Is it because it isn't part of the workspace config of `monorepo_root`?

Reply:
Yes, it is not covered by the workspace globs. As a further detail, different package managers behave differently here, but AFAIK none of them consider this a fully working workspace that is part of the monorepo. Part of the reason it's under the "technically not supported" section.